core/math/quaternion.h

```diff
@@ -141,16 +141,19 @@ struct [[nodiscard]] Quaternion {
 	}
 
 	Quaternion(const Vector3 &p_v0, const Vector3 &p_v1) { // Shortest arc.
-		Vector3 c = p_v0.cross(p_v1);
+		real_t d = p_v0.dot(p_v1);
+		if (Math::is_equal_approx(d, 1.0)) {
```

Comment:
`is_equal_approx()` seems to do some extra unneeded stuff, the old version:
```
if (d < -1.0f + (real_t)CMP_EPSILON)
```
seems better, although I'm not sure it has to be converted to `real_t`, unless we need 64 bit later for the dot (this PR has been refreshed since I initially posted).